experimental/packages/otlp-exporter-base/src/platform/browser/util.ts

```diff
@@ -79,28 +80,37 @@ export function sendWithXhr(
     } else {
       xhr.abort();
     }
-  }, exporterTimeout);
+  }, collector.timeoutMillis);
 
   const sendWithRetry = (
     retries = DEFAULT_EXPORT_MAX_ATTEMPTS,
     minDelay = DEFAULT_EXPORT_INITIAL_BACKOFF
   ) => {
     xhr = new XMLHttpRequest();
-    xhr.open('POST', url);
+    xhr.open('POST', collector.url);
 
     const defaultHeaders = {
       Accept: 'application/json',
-      'Content-Type': 'application/json',
+      'Content-Type': contentType || 'application/json',
     };
 
     Object.entries({
       ...defaultHeaders,
-      ...headers,
+      ...collector.headers,
     }).forEach(([k, v]) => {
       xhr.setRequestHeader(k, v);
     });
 
-    xhr.send(body);
```

Comment:
Architectural design opinion: I don't think loading up code with a bunch of options is good, especially if you want to do this for web use

* For browsers if-s don't get shaken out during bundle time (static analysis doesn't work that deep, mainly `if (false)`-s done by search-and-replace plugins), meaning if someone decides to not use something, it's still a bunch of dead code included in the bundle
* It ties it down to a specific implementation and if there's anything js loves it's a billion ways to do something (eg. pako / deflate / new hotness lib of the month / builtin nodejs gz / compression stream api / maybe someone wants brotli at some point making first point double relevant)

I think it would be better to provide hooks configuration (analogous to middleware `app.use` on the server side) for exporters where callback functions could manipulate request based on users preferences/needs, which is a lot more easily treehshaken (as only functions used would get bundled), eg:

```ts
import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc';

// This is actually where using fetch API internally would allow for a more js standards based API (Request object)
// Also all of this is totally inspired by ky library, gotta throw some respect to well designed libs
function compressUsingPako(req: Request) {
  return new Request(req, {
    body: pako.gzip(req.body),
    headers: {
      'Content-Encoding': 'gzip'
    }
  });
}

const exporter = new OTLPTraceExporter({
  url: 'https://somewhere',
  hooks: [
    compressUsingPako,
    // Hooks could find many other use cases around changes to request
    signRequest,
    attachBatchId,
  ]
});
```